gk/fib.c

```diff
@@ -0,0 +1,1348 @@
+/*
+ * Gatekeeper - DoS protection system.
+ * Copyright (C) 2016 Digirati LTDA.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "gatekeeper_fib.h"
+#include "gatekeeper_gk.h"
+#include "gatekeeper_lls.h"
+#include "gatekeeper_main.h"
+
+void
+destroy_neigh_hash_table(struct neighbor_hash_table *neigh)
+{
+	if (neigh->cache_tbl != NULL) {
+		rte_free(neigh->cache_tbl);
+		neigh->cache_tbl = NULL;
+	}
+
+	if (neigh->hash_table != NULL) {
+		rte_hash_free(neigh->hash_table);
+		neigh->hash_table = NULL;
+	}
+}
+
+static int
+gk_lpm_add_ipv4_routes(struct ipv4_lpm_route *routes,
```

Comment:
`gk_lpm_add_ipv4_routes()` is only called once and for `num_routes == 1`. I think we could just drop `struct ipv4_lpm_route` and have those fields as parameters, and drop parameter `num_routes` and its associated loop. The same goes for the IPv6 case.